src/content_cao.cpp

```diff
@@ -1786,6 +1831,14 @@ void GenericCAO::processMessage(const std::string &data)
 	}
 }
 
+void GenericCAO::knockback(const v3f &direction, f32 time_knockback)
+{
+	m_knockback_timer = time_knockback;
```

Comment:
That coulld be discussed and added in a later PR, let's get the very basic functionality first.